tiled/_tests/test_internal_cache.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+
+from tiled.client import from_config
+from tiled.server.internal_cache import get_internal_cache
+
+
+def test_internal_cache_hit_and_miss(tmpdir):
+    with open(Path(tmpdir, "data.csv"), "w") as file:
+        file.write(
+            """
+a,b,c
+1,2,3
+"""
+        )
+    config = {
+        "trees": [
+            {
+                "tree": "files",
+                "path": "/",
+                "args": {"directory": tmpdir},
+            },
+        ],
+    }
+    client = from_config(config)
+    cache = get_internal_cache()
+    assert cache.hits == cache.misses == 0
+    client["data"]
+    assert cache.misses == 1
+    assert cache.hits == 0
+    client["data"]
+    assert cache.misses == 1
+    assert cache.hits == 1
+    client["data"]
+    assert cache.misses == 1
+    assert cache.hits == 2
```

Comment:
Maybe add a test of the instrumentation of cached? A bug there could send future people on a terrible journey.

Reply:
Cachey has at least [one test](https://github.com/dask/cachey/blob/789e03dbc64e4f39a56724f9b02643c02f704959/cachey/tests/test_cache.py#L99-L119) on this itself, but it's good idea to add an extra layer of protection here since it's important. Will do.

Reply:
Done